dotcom-rendering/src/client/islands/doHydration.tsx

```diff
@@ -26,8 +26,8 @@ export const doHydration = async (
 ): Promise<void> => {
 	// If this function has already been run for an element then don't try to
 	// run it a second time
-	const alreadyHydrated = element.dataset.guReady;
-	if (alreadyHydrated) return;
+	if (element.dataset.island !== undefined) return;
+	else element.dataset.island = 'Importing';
```

Comment:
small thing, but I think it might be helpful to call this `island-status` or something else along those lines.

Also, given we can't(?) add types to this attribute, I think it'd be helpful to add a brief comment to list what statuses we might see, and what they mean.

Reply:
What do you think of the following declaration?

```ts
declare global {
	interface DOMStringMap {
		island?:
			| 'Importing'
			| 'Rendering'
			| 'Hydrating'
			| 'Rendered'
			| 'Hydrated';
	}
}
```

Reply:
If we are reflecting a before and after state then you would also need a 'imported'?

Reply:
I don’t think there’s enough of a different between `hydrating`/`rendering` and `imported`. May be worth using only past tense or present participle:

🎉 – `identified` | `imported` | `rendered` | `hydrated`
🚀 – `importing` | `hydrating` | `rendering` | `reacting`